Added getbundlemetatags policy function (#3402)
cfengine/documentation: reference/functions/getbundlemetatags.markdown

---
layout: default

> title: getclassmetatags

Reviewer: bundle

Reviewer 2: ```suggestion
title: getbundlemetatags
```

published: true
---

[%CFEngine_function_prototype(bundlename, optional_tag)%]

**Description:** Returns the list of [`meta`][Promise types#meta] tags for bundle `bundlename`.

[%CFEngine_function_attributes(bundlename, optional_tag)%]

> The `optional_tag` can be used to look up a specific tag's value. If you format
> your tags like `tags slist => { "mykey=myvalue1", "mykey=myvalue2"}` then the
> `optional_tag` of `mykey` will fetch you a list with two entries, `{ "myvalue1",
> "myvalue2" }`.

Reviewer 2: It might be more clear to show a full meta promise snippet.

```suggestion
The `optional_tag` can be used to look up a specific tag's value. If you have

# Wrap in tickticktickcfengine3
bundle agent example
{
  meta:
    "tags"
      slist => { "mykey=myvalue1", "mykey=myvalue2", "yourkey=yourvalue1" };
}

then `getbundlemetatags( "example", "mykey" )` will return a list with two entries, `{ "myvalue1",
"myvalue2" }`.
```

**Example:**

[%CFEngine_include_snippet(getbundlemetatags.cf, #\+begin_src cfengine3, .*end_src)%]

Output:

[%CFEngine_include_snippet(getbundlemetatags.cf, #\+begin_src\s+example_output\s*, .*end_src)%]

**Notes:**

> **See also:** `getvariablemetatags()`, `getclassemetatags()`

Reviewer: ```suggestion
**See also:** `getvariablemetatags()`, `getclassesmetatags()`
```

---
layout: default
title: getbundlemetatags
published: true
---

[%CFEngine_function_prototype(bundlename, optional_tag)%]

**Description:** Returns the list of [`meta`][Promise types#meta] tags for bundle `bundlename`.

[%CFEngine_function_attributes(bundlename, optional_tag)%]

The `optional_tag` can be used to look up a specific tag's value. If you have

> ```

Reviewer 3: I wonder if we need `cf3` for syntax highlighting :thinking: 
```suggestion
```cf3
```

Reviewer 3: Looks like it's used elsewhere

Reviewer 2: ```cf3 or ```cfengine3 should work IIRC
